docs/manual-setup.md

```diff
@@ -0,0 +1,30 @@
+# Manual Setup Guide
+
+While we strongly encourage you to use the maker command for this bundle, if your use case does not permit you to do so, this guide will outline how your app should interact with the bundle.
+
+The bundle provides a [helper](https://github.com/SymfonyCasts/reset-password-bundle/blob/master/src/ResetPasswordHelper.php) that makes it easy to leverage the bundle to suite your needs. This guide assumes your app will be using this helper. Implementing a customized helper using the [`ResetPasswordHelperInterface`](https://github.com/SymfonyCasts/reset-password-bundle/blob/master/src/ResetPasswordHelperInterface.php) is beyond the scope of this documentation.
+
+## Overview
+
+The bundle has 2 primary responsibilities:
+ - Create a unique, secure, hash for a user that's stored in persistence and provide a non-hashed token to retrieve the stored hash.
+ 
+ - Validate the hash stored in persistence against a user provided token.
+
+The reset password bundle handles these 2 responsibilities very well with a little help from the application using the bundle.
+
+## Creating Tokens
+
+Before a user can create a reset password request, the app must have an entity that implements [`ResetPasswordRequestInterface`](https://github.com/SymfonyCasts/reset-password-bundle/blob/master/src/Model/ResetPasswordRequestInterface.php). This entity will be used by the bundle to store a reset request in persistence. The entity must have a repository that implements [`ResetPasswordRequestRepositoryInterface`](https://github.com/SymfonyCasts/reset-password-bundle/blob/master/src/Persistence/ResetPasswordRequestRepositoryInterface.php). The repository is responsible for the actual creation of the request object and manipulating it  in persistence. 
+
+The bundle provides traits for each of these objects that make it easy to implement the interfaces. Once the above objects have been created in the app, you must set the fully qualified repository class name in the bundle's [configuration](https://github.com/SymfonyCasts/reset-password-bundle/wiki/Confguration-Reference) file.
+
+[`ResetPasswordHelper::generateResetToken()`](https://github.com/SymfonyCasts/reset-password-bundle/blob/239266e8ba6b513c053c86ac51feee9adc4e075c/src/ResetPasswordHelper.php#L63) requires you to provide a user object that will be used to identify the user requesting a reset. After the helper creates a request object and stores it in persistence, a [`ResetPassowrdToken`](https://github.com/SymfonyCasts/reset-password-bundle/blob/master/src/Model/ResetPasswordToken.php) object is returned. The token object contains the token which should be provided to the user for validation later on.
```

Comment:
These sha links aren't going to age well. Well, technically, they will age perfectly! Because they're permanent :). But if the class changes significantly, we will eventually be linking them to old code. Probably the best we can do is link them to the class only on the master branch, and hope they can find their way to the method.

Reply:
Also, type on `ResetPassowrdToken` -> `ResetPasswordToken`

Reply:
> It's kind of annoying, but can you add line breaks on the long lines? We usually add line breaks around the 72-80 char length. It helps with reviewing (tough to post reviews on super long lines).

well that just takes all the fun out of it... haha done. storms md editor is a pain.

Reply:
> These sha links aren't going to age well. Well, technically, they will age perfectly!

Good catch! I didn't even notice the sha's on those two methods.. fixed